modules/osqp-cpp/metadata.json

```diff
@@ -0,0 +1,17 @@
+{
+    "homepage": "https://github.com/google/osqp-cpp",
+    "maintainers": [
+        {
+            "email": "udayaprakash2899@gmail.com",
+            "github": "udaya2899",
+            "name": "Udaya Prakash Nageswaran"
+        }
+    ],
+    "repository": [
+        "github:google/osqp-cpp"
+    ],
+    "versions": [
+        "20231004.0.bcr.1"
```

Comment:
As this repository doesn't have any Git tags, I suggest using the format `0.0.0-YYYYMMDD-SHORTSHA` (`0.0.0-20240530-2db0eb3`) as is done for other modules.